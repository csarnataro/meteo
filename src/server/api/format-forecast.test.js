const { formatForecast } = require('./format-forecast')
const data = require('./__mocks__/api-darksky-forecast')

describe('formatForecast', () => {
  it('should format', async () => {
    const output = await formatForecast(data)
    expect(output).not.toBeUndefined()
    expect(output).toMatchSnapshot()
  })
})
