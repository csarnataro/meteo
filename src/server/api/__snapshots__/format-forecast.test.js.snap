// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatForecast should format 1`] = `
Object {
  "daily": Array [
    Object {
      "date": "Mercoledì     14     Agosto,     00:00",
      "description": "Poco nuvoloso durante il giorno.",
      "icon": "partly-cloudy-day",
    },
    Object {
      "date": "Giovedì     15     Agosto,     00:00",
      "description": "Poco nuvoloso durante il giorno.",
      "icon": "partly-cloudy-day",
    },
    Object {
      "date": "Venerdì     16     Agosto,     00:00",
      "description": "Nubi sparse durante il giorno.",
      "icon": "partly-cloudy-day",
    },
    Object {
      "date": "Sabato     17     Agosto,     00:00",
      "description": "Nubi sparse durante il giorno.",
      "icon": "partly-cloudy-day",
    },
    Object {
      "date": "Domenica     18     Agosto,     00:00",
      "description": "Sereno durante il giorno.",
      "icon": "clear-day",
    },
    Object {
      "date": "Lunedì     19     Agosto,     00:00",
      "description": "Nubi sparse durante il giorno.",
      "icon": "partly-cloudy-day",
    },
    Object {
      "date": "Martedì     20     Agosto,     00:00",
      "description": "Nubi sparse durante il giorno.",
      "icon": "rain",
    },
    Object {
      "date": "Mercoledì     21     Agosto,     00:00",
      "description": "Possibilità di pioggia leggera nella notte.",
      "icon": "rain",
    },
  ],
  "date": "Mercoledì     14     Agosto,     09:22",
  "hourly": Array [
    Object {
      "date": "Mercoledì     14     Agosto,     09:00",
      "description": "Poco Nuvoloso",
      "icon": "partly-cloudy-day",
    },
    Object {
      "date": "Mercoledì     14     Agosto,     12:00",
      "description": "Nubi Sparse",
      "icon": "partly-cloudy-day",
    },
    Object {
      "date": "Mercoledì     14     Agosto,     15:00",
      "description": "Poco Nuvoloso",
      "icon": "partly-cloudy-day",
    },
    Object {
      "date": "Mercoledì     14     Agosto,     18:00",
      "description": "Sereno",
      "icon": "clear-day",
    },
    Object {
      "date": "Mercoledì     14     Agosto,     21:00",
      "description": "Sereno",
      "icon": "clear-night",
    },
    Object {
      "date": "Giovedì     15     Agosto,     00:00",
      "description": "Sereno",
      "icon": "clear-night",
    },
    Object {
      "date": "Giovedì     15     Agosto,     03:00",
      "description": "Sereno",
      "icon": "clear-night",
    },
  ],
  "icon": "partly-cloudy-day",
  "location": "Milano",
}
`;
